### ISSUE TITLE
error with releases task

### ISSUE THREAD

**Issue by jspaaks** [research-software-directory/research-software-directory#178]
```
tasks_1      | 2018-11-29 10:12:22,792 releases [INFO] 97/124 "Emma": 10.5281/zenodo.996294 no valid CITATION.cff found in any release.
tasks_1      | 2018-11-29 10:12:24,961 releases [INFO] 98/124 "ShiCo": 10.5281/zenodo.1187089 OK
tasks_1      | 2018-11-29 10:12:27,311 releases [INFO] 99/124 "SIGA.py": 10.5281/zenodo.1076437 no valid CITATION.cff found in any release.
tasks_1      | 2018-11-29 10:12:29,558 releases [INFO] 100/124 "SPOT": 10.5281/zenodo.1003345 no valid CITATION.cff found in any release.
tasks_1      | 2018-11-29 10:12:32,332 releases [INFO] 101/124 "StoryTeller": 10.5281/zenodo.996323 no valid CITATION.cff found in any release.
tasks_1      | Traceback (most recent call last):
tasks_1      |   File "app.py", line 62, in <module>
tasks_1      |     run_task()
tasks_1      |   File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
tasks_1      |     return self.main(*args, **kwargs)
tasks_1      |   File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
tasks_1      |     rv = self.invoke(ctx)
tasks_1      |   File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
tasks_1      |     return ctx.invoke(self.callback, **ctx.params)
tasks_1      |   File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
tasks_1      |     return callback(*args, **kwargs)
tasks_1      |   File "app.py", line 53, in run_task
tasks_1      |     sync_releases(db)
tasks_1      |   File "/app/releases.py", line 202, in sync_releases
tasks_1      |     for item_index, software_item in enumerate(software_items):
tasks_1      |   File "/usr/local/lib/python3.6/site-packages/pymongo/cursor.py", line 1169, in next
tasks_1      |     if len(self.__data) or self._refresh():
tasks_1      |   File "/usr/local/lib/python3.6/site-packages/pymongo/cursor.py", line 1106, in _refresh
tasks_1      |     self.__send_message(g)
tasks_1      |   File "/usr/local/lib/python3.6/site-packages/pymongo/cursor.py", line 975, in __send_message
tasks_1      |     helpers._check_command_response(first)
tasks_1      |   File "/usr/local/lib/python3.6/site-packages/pymongo/helpers.py", line 142, in _check_command_response
tasks_1      |     raise CursorNotFound(errmsg, code, response)
tasks_1      | pymongo.errors.CursorNotFound: cursor id 52379082769 not found
mongo_1      | 2018-11-29T10:12:32.655+0000 I NETWORK  [conn25] end connection 172.24.0.3:41886 (13 connections now open)
mongo_1      | 2018-11-29T10:12:32.655+0000 I NETWORK  [conn20] end connection 172.24.0.3:40672 (12 connections now open)
mongo_1      | 2018-11-29T10:12:32.668+0000 I NETWORK  [conn15] end connection 172.24.0.3:40664 (11 connections now open)
mongo_1      | 2018-11-29T10:12:32.668+0000 I NETWORK  [conn21] end connection 172.24.0.3:40676 (10 connections now open)
mongo_1      | 2018-11-29T10:12:32.668+0000 I NETWORK  [conn23] end connection 172.24.0.3:41248 (9 connections now open)
mongo_1      | 2018-11-29T10:12:32.669+0000 I NETWORK  [conn24] end connection 172.24.0.3:41412 (8 connections now open)
mongo_1      | 2018-11-29T10:12:32.669+0000 I NETWORK  [conn18] end connection 172.24.0.3:40670 (7 connections now open)
backend_1    | 172.24.0.7 - - [29/Nov/2018:10:14:33 +0000] "GET /api/software_cache/spot HTTP/1.1" 200 167312 "-" "python-requests/2.20.1"
frontend_1   | 172.24.0.8 - - [29/Nov/2018:10:14:33 +0000] "GET /software/spot?asdasd223 HTTP/1.0" 200 167921 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36"
nginx_1      | 172.24.0.5 - - [29/Nov/2018:10:14:33 +0000] "GET /software/spot?asdasd223 HTTP/1.0" 200 167921 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36" "172.24.0.1"
nginx_ssl_1  | 172.24.0.1 - - [29/Nov/2018:10:14:33 +0000] "GET /software/spot?asdasd223 HTTP/1.1" 200 167921 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36" "-"
```

**Comment by jspaaks** [comment]
possibly has something to do with concurrent tasks ?

**Comment by Tommos0** [comment]
Hmm it's not running concurrently right?

**Comment by jspaaks** [comment]
this was just a note for future reference. I was fiddling with the crontab. I collected all jobs in one ``python app.py --task all`` task, and then running that at 1hr intervals, such that engineers get feedback more quickly.  But simultaneously to the ``all`` task, there is still the ``cache_software`` task. I've seen examples where the output is interleaved, but with the output above something went wrong.

In retrospect, I think maybe this could have been caused by https://github.com/research-software-directory/research-software-directory/issues/180 (which is now fixed)

If we don't see this error for a while, I'm fine with called this one fixed as well.
